components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java

```diff
@@ -251,8 +251,15 @@ public Repository create(final Configuration configuration) throws Exception {
 
     log.debug("Creating repository: {} -> {}", repositoryName, configuration);
 
-    Repository repository = newRepository(configuration);
-    store.create(configuration);
+    Repository repository;
+    try {
+      store.create(configuration);
+      repository = newRepository(configuration);
+    }
+    catch (Exception e) {
+      store.delete(configuration);
+      throw e;
+    }
```

Comment:
To align how repositories are created: default ones had config _saved_ then _created_, newly added one (w/o change above), were _created_ then the config _saved_. This meant that in doConfigure and related methods in one case the Configuration was equipped with entity metadata, in other was not. This prevented use of ConfigFacet.save method in case of newly created reposes, while it did work for default ones.